App/app.rb

```diff
@@ -0,0 +1,118 @@
+  def initialize(parent)
+    @parent = parent
+    @books_list = []
+    @people_list = []
+    @rentals_list = []
+  end
+
+  def list_all_books
+    if @books_list.empty?
+      puts 'No record found! Add some books...'
+    else
+      puts 'Available books in the library'
+      @books_list.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    end
+    @parent.show_menu
+  end
+
+  def list_all_people
+    if @people_list.empty?
+      puts 'No record found! Add a person...'
+    else
+      puts 'All people in the library'
+      @people_list.each do |person|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+    @parent.show_menu
+  end
+
+  def create_person
+    puts 'Do you want to create a student(1) or a teacher(2)? [Input the number]'
+    person_role = gets.chomp
+    case person_role
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Please add a valid input!'
+    end
+    @parent.show_menu
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase == 'y'
+    print 'Classroom: '
+    classroom = gets.chomp.to_i
+    @people_list.push(Student.new(age, classroom, name, parent_permission: parent_permission))
+    puts
+    puts 'Person created successfuly'
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    @people_list.push(Teacher.new(age, specialization, name))
+    puts
+    puts 'Person created successfuly'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books_list.push(Book.new(title, author))
+    puts 'Book created successfully'
+    @parent.show_menu
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    @books_list.each_with_index { |book, index| puts "#{index}) Title: '#{book.title}', Author: #{book.author}" }
+    book_num = gets.chomp.to_i
+    puts
+    puts 'Select a person from the following list by number (not id)'
+    @people_list.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_num = gets.chomp.to_i
+
+    print 'Date: '
+    date = gets.chomp
+    @rentals_list.push(Rental.new(date, @books_list[book_num], @people_list[person_num]))
+    puts 'Rental created successfully'
+    @parent.show_menu
+  end
+
+  def list_all_rentals
+    print 'ID of person: '
+    id = gets.chomp.to_i
+
+    puts 'Rentals:'
+    @rentals_list.each do |rental|
+      puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}" if rental.person.id == id
+    end
+    @parent.show_menu
+  end
```

Comment:
**[Optional]**

- [ ] It would be very nice if you show the id's of all the people from getting the id as input to show the rentals. Because when i would use this option then i would need to input the id of the person whom rental books i wanna see so how i am gonna find out which id belongs to which student or teacher. So it would be nice if you print people with their id's before taking id as an input 😊